test/orderController.test.js

```diff
@@ -0,0 +1,132 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Order = require('../src/models/Order');
+const User = require('../src/models/User');
+const Album = require('../src/models/Album');
+const { loginAndReturnCookie } = require('./test_helper');
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('OrderController API Tests', () => {
+  let user;
+  let album;
+  let userCredentials;
+  let response;
+
+  beforeAll(async () => {
+    //create a user
+    user = await User.create({
+      email: 'Emily@google.com',
+      password: 'secret',
+      name: 'Emily',
+      username: 'emily123',
+      role: 'user',
+    });
+    // Create an album
+    album = await Album.create({
+      albumName: 'Unique Album',
+      artistName: 'Unique Artist',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/unique',
+    });
+  });
+
+  beforeEach(async () => {
+    userCredentials = {
+      email: 'Emily@google.com',
+      password: 'secret',
+    };
+  });
+
+  //create an order -Success case
+  it('should create an order successfully', async () => {
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+    const mockPaymentIntent = {
+      id: 'pi_mock123',
+      client_secret: 'mock_secret',
+    };
+    //data for order
+    let orderData = {
+      orderItems: [
+        {
+          album: album._id,
+          quantity: 2,
+        },
+      ],
+      subtotal: 100,
+      tax: 10,
+      total: 110,
+      user: user._id,
+      paymentIntentId: mockPaymentIntent.id,
```

Comment:
When the client calls this "Create Order" endpoint, they won't have a payment intent id yet. It's only in this `createOrder` controller that the backend _first_ creates a payment intent instance for this order. 

Trying to pass one in here, will just be ignored by the controller since there's not code in there to parse that field out of the body: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/67cb52b6279196f8fea68ba34dd27db8680ff0d8/src/controllers/orderController.js#L15](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/67cb52b6279196f8fea68ba34dd27db8680ff0d8/src/controllers/orderController.js#L15)

Currently, our orderController uses a test stripe API key, so it will just be making the real request to Stripe as part of this test, using a test key. If we had a real live production key, we of course would not want our tests to be creating real payement intents with that key; so we would set the API key to the test key in the `beforeAll` block, similar to what we do to set the `MONGO_URL` to point to the test database.

Another approach, if we don't want to be actually making requests to Stripe (eg: if we feel it adds to much latency/time to the tests, or if we're concerned that our Github Action CI or somewhere else that we're running the tests won't have access to the network/internet) then we could just [mock](https://spin.atomicobject.com/2021/06/28/mocks-stubs-or-spies/) the stripe call altogether:

```javascript
# This goes either at the very top of your test file (if you want to use the same mock for all tests) or at the top of particular test case where you want this mock
jest.mock('stripe', () => {
  // When `require('stripe')` is called, the below jest mock function will be returned instead
  return jest.fn(() => ({
    // When the `require('stripe')(API_KEY)` is called, the below object will be returned instead
    paymentIntents: {
      // The `stripe.paymentIntents.create` will be replaced with the below jest mock function that returns our mock data
      create: jest.fn(() => ({
        id: 'pi_mock123',
        client_secret: 'mock_secret',
      }))
    }
  }))
});
```